addons/sale/models/sale.py

```diff
@@ -1716,7 +1716,16 @@ def product_id_change(self):
         self._compute_tax_id()
 
         if self.order_id.pricelist_id and self.order_id.partner_id:
-            vals['price_unit'] = self.env['account.tax']._fix_tax_included_price_company(self._get_display_price(product), product.taxes_id, self.tax_id, self.company_id)
+            vals['price_unit'] = product._get_tax_included_unit_price(
+                self.company_id or self.order_id.company_id,
```

Comment:
but if we consider that https://github.com/odoo/odoo/pull/78032 was not forward ported for a reason, then this line should be

```suggestion
                self.company_id,
```
and same below